controllers/update_test.go

```diff
@@ -872,10 +875,39 @@ func compareRepoWithExpected(g *WithT, repoURL, branch, fixture string, changeFi
 	test.ExpectMatchingDirectories(g, actual, expected)
 }
 
-func commitInRepo(g *WithT, repoURL, branch, msg string, changeFiles func(path string)) {
-	originRemote := "origin"
-	repo, err := clone(repoURL, originRemote, branch)
+// configureManagedTransportOptions registers the transport options for this repository
+// and sets the remote url of origin to the transport options url. If repoURL is empty
+// it tries to figure it out by looking at the remote url of origin. It returns a function
+// which removes the transport options for this repo and sets the remote url of the origin
+// back to the actual url. Callers are expected to call this function in a deferred manner.
+func configureManagedTransportOptions(repo *libgit2.Repository, repoURL string) (func(), error) {
+	if repoURL == "" {
+		origin, err := repo.Remotes.Lookup(originRemote)
+		if err != nil {
+			return nil, err
+		}
+		defer origin.Free()
+		repoURL = origin.Url()
+	}
```

Comment:
Based on the usage of this function, it looks like it'd be better to not accept `repoURL` and derive it out of the `repo` always.

Reply:
And with that, this function can be renamed to `configureManagedTransportOptionsForRepo` or something short and better if you can think of.

Reply:
It may be the case that the origin URL lacks http credentials, that should be verified. But if that's the case, the `repoURL` argument can be kept and renamed appropriately, maybe `urlWithCreds` or something better.